packages/compose/src/hooks/use-focus-return/index.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import { useRef, useCallback, useEffect } from '@wordpress/element';
+
+/**
+ * When opening modals/sidebars/dialogs, the focus
+ * must move to the opened area and return to the
+ * previously focused element when closed.
+ * The current hook implements the returning behavior.
+ *
+ * @param {Function?} onFocusReturn Overrides the default return behavior.
+ * @return {Function} Element Ref.
+ *
+ * @example
+ * ```js
+ * import { useFocusReturn } from '@wordpress/compose';
+ *
+ * const WithFocusReturn = () => {
+ *     const ref = useFocusReturn()
+ *     return (
+ *         <div ref={ ref }>
+ *             <Button />
+ *             <Button />
+ *         </div>
+ *     );
+ * }
+ * ```
+ */
+function useFocusReturn( onFocusReturn ) {
+	const focusedBeforeMount = useRef();
+	const isFocused = useRef( false );
+	const onFocusReturnRef = useRef( onFocusReturn );
+	useEffect( () => {
+		onFocusReturnRef.current = onFocusReturn;
+	}, [] );
+
+	const ref = useCallback( ( newNode ) => {
+		const updateLastFocusedRef = ( { target } ) => {
+			isFocused.current = newNode && newNode.contains( target );
+		};
+
+		// Unmounting the reference
+		if ( ! newNode && focusedBeforeMount.current ) {
+			if ( newNode?.ownerDocument ) {
```

Comment:
How can `newNode.ownerDocument` be defined if previously there's a check for `! newNode`?